src/Content/Text/BBCode.php

```diff
@@ -1727,10 +1727,10 @@ function ($matches) use ($simple_html) {
 			$text = Smilies::replace($text);			
 		}
 
-		if (DI::config()->get('system', 'big_emojis') && ($simple_html != self::DIASPORA)) {
+		if (!$for_plaintext && DI::config()->get('system', 'big_emojis') && ($simple_html != self::DIASPORA)) {
```

Comment:
Why do we need this parameter, aren't we applying `HTML::toPlaintext` afterwards? Wouldn't it take care of the added HTML tag?

Reply:
Why should we add some HTML, just to remove it in the next step?

Reply:
Because this method is meant to output HTML. Most of its current uses overlap with other flags (simplehtml, nosmilies) and it should be removed entirely for simplicity over a negligible performance hit.

Reply:
When - at some time - that parameter will be replaced, then using it here will be a good reminder. I don't like adding stuff that is removed in the next step. This adds unneeded complexity and opens the doors for side effects. 

Indeed I can imagine that the `simple_html` parameter can be used for that as well. I can have a look in the next days.